File: leetcode/strings/1816. Truncate Sentence/python/main.py
class Solution:
    def truncateSentence(self, s: str, k: int) -> str:
        spaces, tar = 0, " "
        for i in range(len(s)):
            if s[i] == tar:
                spaces += 1
            if spaces == k:
                return s[:i]
        return s


s = "What is the solution to this problem"
k = 4
print(Solution().truncateSentence(s, k))

File: leetcode/strings/1816. Truncate Sentence/golang/main.go
package main

import (
	"fmt"
)

func truncateSentence(s string, k int) string {
	spaces, target := 0, ' '
	for i, v := range s {
		if v == target {
			spaces++
		}
		if spaces == k {
			return s[:i]
		}
	}
	return s
}

func main() {
	s := "What is the solution to this problem"
	k := 4
	fmt.Println(truncateSentence(s, k))

}
